### ISSUE TITLE
Version conflicts sharing a PCL between ASP.NET 5.0 and UWP

### ISSUE THREAD

**Issue by awkoren** [aspnet/Tooling#442]
I'm trying to share some model classes between an ASP.NET 5.0 project (using framework DNX 4.5.1 only for now) and a UWP app. However, I'm running into versioning troubles with the ObservableCollection class. This is on VS 2015 Update 2 (but also occurred on Update 1). 

If I use a regular PCL, the ASP.NET 5.0 project gives this error: 

```
The type 'ObservableCollection<>' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.ObjectModel, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
```

This will not go away even when I add System.ObjectModel 4.0.0.0 to the ASP.NET 5.0 project.json file. Adding a reference to the DLL directly at C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.1\Facades *appears* to resolve the problem at first, but then starts throwing cryptic "Build failed" errors when I tried to publish the project.  

So I tried using the new Class Library (Package) type. However, if I try to add the nuget package to the UWP app, it says:  

```
Install-Package : Version conflict detected for System.Collections. 
 App (>= 1.0.0) -> Models (>= 1.0.0) -> System.Collections 
(>= 4.0.11-beta-23516) 
 App (>= 1.0.0) -> Microsoft.NETCore.UniversalWindowsPlatform (>= 
5.0.0) -> Microsoft.NETCore.Runtime (>= 1.0.0) -> 
Microsoft.NETCore.Runtime.CoreCLR-arm (>= 1.0.0) -> System.Collections (= 
4.0.10).
At line:1 char:1
+ Install-Package Models
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Install-Package], Exception
    + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PackageManag 
   ement.PowerShellCmdlets.InstallPackageCommand
```
I tried specifying a reference to System.Collections 4.0.11-beta-23516 in my UWP app, but this only brought more conflicts. 

In short, regular PCL works with UWP but not ASP.NET 5.0, while Nuget PCL works with ASP.NET 5.0 but not UWP.

Any suggestions on how to resolve this and make everything play nicely?

**Comment by sayedihashimi** [comment]
@awkoren we are adding a feature to allow xproj project to reference csproj projects, which should be in the next release. I think that should hopefully solve these issues right?

cc @terrajobst

**Comment by awkoren** [comment]
@sayedihashimi Can you elaborate? I can reference a .csproj PCL today from my ASP.NET .xproj site ... it's just the one DLL causing problems ... so not clear to me how this feature fixes this. Also, will the feature work in reverse (.csproj reference .xproj)?

**Comment by sayedihashimi** [comment]
@awkoren 
>I can reference a .csproj PCL today from my ASP.NET .xproj site .

How are you adding the reference?

Yes csproj will be able to reference xproj. There will be some limitations (especially in our first release) but it should be possible.

**Comment by awkoren** [comment]
@sayedihashimi Right click references -> Add reference -> Solution. Then choose a .csproj PCL project and it adds just fine.

**Comment by sayedihashimi** [comment]
@awkoren yes now I remember. We used "dnu wrap" for that, which has significant limitations. We have redone the implementation and it should be in the next release. Can you share your solution so that I can check to see if this has been fixed?

**Comment by synestral** [comment]
I see lots of people having these issues, myself included, and always the same answer, "will be fixed in RC2"...considering RC2 could be (and probably is) months away from release, i have to wonder how these things have gone ignored for so long :(

ASP.NET Project can't reference PCL, UWP Apps can't reference the DNX Libraries (the class library package one)... I don't even want to think about how many developers are being blocked by these things, hell of a situation to be stuck in with no "fix" available for possibly months...

**Comment by sayedihashimi** [comment]
@synestral I understand that it's frustrating to work with bits which are not completed. In this case our RC2 release has been delayed, which is exacerbating this. We had hoped to get RC2 out the door earlier but unfortunately we were not able to do that. We are working hard to get the release out the door. Once we release our RTM bits it will be much easier for us to ship updates. Right now there is a lot of churn across a lot of areas. Unfortunately we can't release a fix for specific issues now because it would require releasing a bunch of updates.

Over the past few years the ASP.NET team has made investments to enable us to ship updates faster. This is the reason why we can ship without requiring a Visual Studio update for many scenarios. We've made good progress there, but we still have a ways to go. It's something that we are interested in and hopefully we can do a better job going forward.

Regarding this specific issue. We initially added support for adding references via `dnu wrap`. After releasing that we received a good amount of feedback that it wasn't meeting the need of customers. That's why we have decided to revisit this and to do a better job. I'm hoping that the updates that we are working on for RC2 will improve this. The issue has not been ignored, we are actively working on it. I'm sorry you're running into issues, we are doing our best to be transparent and to get fixes out as soon as possible.

**Comment by sayedihashimi** [comment]
@synestral we have released RC2, can you try with RC2 and let me know if this is still an issue? I'll close this for now but please reopen if we need to address it.
